src/app/pipes/alm-valid-link-types.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { WorkItem } from '../models/work-item';
+
+@Pipe({ name: 'almValidLinkTypes', pure: true })
+export class AlmValidLinkTypes implements PipeTransform {
+  transform(arr: any[], workItem: WorkItem): any {
+    let opLinkTypes: any[] = [];
+    arr.forEach((item) => {
+      if (item.relationships.source_type.data.id == workItem.relationships.baseType.data.id) {
+        opLinkTypes.push({
+          name: item.attributes['forward_name'],
+          linkId: item.id,
+          linkType: 'forward'
+        });
+      } else if (item.relationships.target_type.data.id == workItem.relationships.baseType.data.id){
```

Comment:
Well, by the logic they has to match either one of them. If they are not then it's a bug in linking.